### ISSUE TITLE
space-before-function-parentheses error

### ISSUE THREAD

**Issue by kentor** [babel/babel-eslint#38]
I have the rule `space-before-function-parentheses: [2, never]` which gives an error when there's a space before the function keyword:

```javascript
function () {}
```

But this is also giving the same error:
```javascript
[1].map(x => x);
```

**Comment by sebmck** [comment]
Impossible to fix since it'd require injecting tokens for ESLint to consume. Try upgrading to ESLint 0.16.

**Comment by kentor** [comment]
Sorry. I failed to mention I was using eslint 0.16

I only get the space-before-function-parentheses error when using babel-eslint as the parser. If I just use the default parser that comes with eslint I don't get it.

index.js:
```javascript
[1].map(x => x);
```

package.json:
```json
{
  "name": "eslint-test",
  "version": "0.0.0",
  "devDependencies": {
    "babel-eslint": "^1.0.14",
    "eslint": "^0.16.1"
  }
}
```

.eslintrc
```yaml
---
parser: babel-eslint # take this out and it'll pass
ecmaFeatures:
  arrowFunctions: true
rules:
  space-before-function-parentheses: [2, never]
  strict: 0
```

```
$ ./node_modules/.bin/eslint index.js

index.js
  1:9  error  Unexpected space before function parentheses  space-before-function-parentheses

✖ 1 problem (1 error, 0 warnings)
```
